File: AzurposService/app/src/main/java/merryweather/com/azurposservice/di/modules/ActivityModule.java
package merryweather.com.azurposservice.di.modules;

import android.app.Activity;
import android.content.Context;


import dagger.Module;
import dagger.Provides;
import merryweather.com.azurposservice.di.scopes.PerActivity;


@Module
public class ActivityModule {

    private final Activity activity;

    public ActivityModule(Activity activity) {
        this.activity = activity;
    }

    @Provides
    @PerActivity
    Context provideContext() {
        return activity;
    }

}

File: Nista/app/src/main/java/com/example/nista/di/modules/ActivityModule.kt
package com.leroymerlin.cds.di.modules

import android.app.Activity
import android.content.Context

import dagger.Module
import dagger.Provides
import com.leroymerlin.cds.di.scopes.PerActivity

@Module
class ActivityModule(private val activity: Activity) {

    @Provides
    @PerActivity
    internal fun provideContext(): Context {
        return activity
    }

}